backend/generator.js

```diff
@@ -45,18 +47,26 @@ exports.executeScript = function (socket, formData, callback) {
   socketHandler.handleLineError(socket, process, 'err-logs');
 
   process.on('exit', function (code) {
+    var reponame = gitUrl.split('/')[3] + '/' + gitUrl.split('/')[4].split('.')[0];
```

Comment:
This is done multiple times from different files. Can this be defined in a single function and be called from different places?